src/Piling/PilingUpdater.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Autodesk.Revit.DB;
+using System;
+using System.Collections.Generic;
+using Autodesk.Revit.ApplicationServices;
+using Jpp.Cedar.Core;
+
+namespace Jpp.Cedar.Piling
+{
+    public class PilingUpdater : IUpdater
+    {
+        AddInId _appId;
+        UpdaterId _updaterId;
+
+        ISharedParameterManager _spManager;
+
+        bool registered = false;
+        Definition _eastingDefinition, _northingDefinition, _cutOffDefinition;
+
+        public PilingUpdater(AddInId id, ISharedParameterManager spManager)
+        {
+            if (id == null)
+                throw new System.ArgumentNullException(nameof(id));
+
+            if (spManager == null)
+                throw new System.ArgumentNullException(nameof(spManager));
+
+            _appId = id;
```

Comment:
Consider single line join and check.

_appId = id ?? throw new System.ArgumentNullException(nameof(id));

Reply:
Feels like _appId is just local to the constructor?